### PR TITLE
[flutter_tools] shard out two integration tests we want to run on macOS arm64

### PR DESCRIPTION
**Pull request by christopherfujino**
Moved one test from integration.shard/bash_entrypoint_test.dart and the complete file of integration.shard/macos_content_validation_test.dart to a new shard, mac_arm.shard/. Also created a new builder in .ci.yaml, copying `Mac tool_integration_tests 1_4`, while deleting the goldctl dependency.

### CODE REVIEW

**Review by jmagman — COMMENTED** [review]
I don't think `mac_arm.shard` makes sense, these tests need to run on x64 as well...
Should also include `ios_content_validation_test` for the simulator arch tests.

**Review by keyonghan — COMMENTED** [review]
Config LGTM, but we need to add the test ownership to pass the test.

**Comment by christopherfujino** [comment]
> Config LGTM, but we need to add the test ownership to pass the test.

Should this be added as a comment near the end of TESTOWNERS? I'm not sure what's going on in that file.

**Comment by keyonghan** [comment]
> > Config LGTM, but we need to add the test ownership to pass the test.
> 
> Should this be added as a comment near the end of TESTOWNERS? I'm not sure what's going on in that file.

That's right.

**Comment by christopherfujino** [comment]
cc @Jasguerrero for context, regarding our 1:1 conversation, this is what it currently takes to create a new test shard.

File: .ci.yaml [hunk 1]
```diff
@@ -2913,6 +2913,31 @@ targets:
...
+    recipe: flutter/flutter_drone
+    bringup: true
+    timeout: 60
+    properties:
+      add_recipes_cq: "true"
+      dependencies: >-
+        [
+          {"dependency": "android_sdk", "version": "version:31v8"},
+          {"dependency": "chrome_and_driver", "version": "version:98.1"},
+          {"dependency": "open_jdk", "version": "11"},
```
[the remark this answers is not in the document]
Comment (christopherfujino) [reply .ci.yaml:2926]: SGTM, i already removed goldctl

**Review by jmagman — APPROVED** [review]
LGTM, thanks Chris

**Comment by christopherfujino** [comment]
New builder was created but no builds scheduled yet. For monitoring: https://ci.chromium.org/p/flutter/builders/staging/Mac%20tool_host_cross_arch_tests

**Comment by christopherfujino** [comment]
First seven builds all green, looks like a success https://ci.chromium.org/p/flutter/builders/staging/Mac%20tool_host_cross_arch_tests

**Pull request merged**